src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1020,6 +1022,27 @@ protected override void HandleInitializeRequestAsync(IRequestResponder<Initializ
             responder.SetResponse(initializeResponse);
         }
 
+        private bool m_canReverse = false;
+        private void UpdateCapabilities()
+        {
+            bool canReverse = ((IDebugReversibleEngineProgram160)m_engine).CanReverse() == HRConstants.S_OK;
```

Comment:
>((IDebugReversibleEngineProgram160)m_engine [](http://example.com/codeflow?start=30&length=43)

We should handle the case that the engine doesn't implement this, and we should use `m_program` instead of `m_engine`.